app/assets/javascripts/discourse/app/routes/discovery.js

```diff
@@ -60,4 +61,22 @@ export default class DiscoveryRoute extends DiscourseRoute {
   triggerRefresh() {
     this.refresh();
   }
+
+  activate(transition) {
+    super.activate(...arguments);
+
+    if (transition?.to.name === "discovery.custom") {
```

Comment:
This doesn't work reliably because `activation/deactivation` doesn't get triggered when moving to other discovery routes.